### ISSUE TITLE
JavaScript.Web.MessageEvent.getData broken

### ISSUE THREAD

**Issue by stepcut** [ghcjs/ghcjs-base#60]
getData seems to have `2 == BlobData` and `3 == ArrayBuffer`:

```
getData :: MessageEvent -> MessageEventData
getData me = case js_getData me of
               (# 1#, r #) -> StringData      (JSString r)
               (# 2#, r #) -> BlobData        (SomeBlob r)
               (# 3#, r #) -> ArrayBufferData (SomeArrayBuffer r)
{-# INLINE getData #-}
```

but the `js_getData` FFI import seems to have `2 == ArrayBuffer` and `3 == BlobData`

```
foreign import javascript unsafe
  "$r2 = $1.data;\
  \$r1 = typeof $r2 === 'string' ? 1 : ($r2 instanceof ArrayBuffer ? 2 : 3)"
  js_getData :: MessageEvent -> (# Int#, JSVal #)

```

https://github.com/ghcjs/ghcjs-base/blob/master/JavaScript/Web/MessageEvent.hs

**Comment by DougBurke** [comment]
Is this fixed by #54?

**Comment by stepcut** [comment]
I can't easily test it -- but it sure looks like it should.

**Comment by tittoassini** [comment]
I have wasted hours because of this bug, why hasn't the fix been merged in?

**Comment by bergey** [comment]
@stepcut @DougBurke @tittoassini Can one of you confirm that this is fixed in a recent build (since #54 was merged)?

**Comment by tittoassini** [comment]
yes, it is fixed
